blob/fileblob/fileblob.go

```diff
@@ -606,7 +617,104 @@ func (b *bucket) Delete(ctx context.Context, key string) error {
 	return nil
 }
 
+// SignedURL takes a context, an object key, and a SignedURLOptions struct
+// and returns a string containing a URI and an error
 func (b *bucket) SignedURL(ctx context.Context, key string, opts *driver.SignedURLOptions) (string, error) {
-	// TODO(Issue #546): Implemented SignedURL for fileblob.
-	return "", errNotImplemented
+	if b.opts.URLSigner == nil {
+		return "", errNotImplemented
+	}
+	return b.opts.URLSigner.URLFromKey(ctx, key, opts)
+}
+
+// URLSigner defines an interface for
+// creating and verifying a signed URL for objects
+// in a fileblob bucket. Signed URLs are typically used for
+// granting access to an otherwise-protected resource without
+// requiring further authentication, and callers should take care
+// to restrict the creation of signed URLs as is appropriate
+// for their application.
+type URLSigner interface {
+	// URLFromKey defines how the bucket's object key will be turned
+	// into a signed URL.
+	// URLFromKey takes in a context, object key and options.
+	// It returns a signed URL string, and an error.
+	URLFromKey(ctx context.Context, key string, opts *driver.SignedURLOptions) (string, error)
+
+	// KeyFromURL takes in a context and a URL.
+	// It returns an object key string and a bool.
+	// e.g. (key, true) if the signature is valid (authentic && unexpired),
+	// or (nil, false) if it is invalid.
+	// KeyFromURL must be able to validate a URL from the URLFromKey function.
+	KeyFromURL(ctx context.Context, surl *url.URL) (string, bool)
```

Comment:
Nit: it's a bit weird to say that the output from one function can be passed to another function, but have them be different types. Can they both be `string` or both be `url.URL`?

Reply:
I originally had them both as `string`, since the `SignedURL` signature returns a string, but the request URL is a `url.URL` object already, so to make that one take a string you end up converting it to a string and then re-parsing it, which seems sub-optimal.

Reply:
made them both `url.URL`